### DISCUSSION TITLE
Does the editor support more features，Does it support uploading images to Alibaba OSS?

### DISCUSSION THREAD

**Discussion by DreamPWJ** [tinacms/tinacms#3961] [General]
Is the editor function still very simple now，  Does it support uploading multiple images？ Does it support complex editing and video uploading?

**Comment by logan-anderson** [comment]
Hello @DreamPWJl, the default media store does support uploading multiple images to your repo. You can not edit media (you would have to do that before uploading).

We currently support
- [Cloudinary](https://tina.io/docs/reference/media/external/cloudinary/)
- [S3](https://tina.io/docs/reference/media/external/s3/)
- [Digital Ocean Spaces](https://tina.io/docs/reference/media/external/do-spaces/)

You could make your own media store and use any service you like.
